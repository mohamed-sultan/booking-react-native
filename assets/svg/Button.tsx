import * as React from 'react';
import Svg, { SvgProps, Path } from 'react-native-svg';
import { memo } from 'react';
const SvgComponent = (props: SvgProps) => {
  const { width = 40, height = 40, color = '#1E1E22', ...rest } = props;
  return (
    <Svg width={width} height={height} fill="none" {...rest}>
      <Path
        fill={color}
        fillOpacity={0.05}
        d="M0 12C0 5.373 5.373 0 12 0h16c6.627 0 12 5.373 12 12v16c0 6.627-5.373 12-12 12H12C5.373 40 0 34.627 0 28V12Z"
      />
      <Path
        fill={color}
        fillRule="evenodd"
        d="M25.637 25.637a1.109 1.109 0 0 1-1.646-.006l-3.974-3.974-4.008 3.975a1.108 1.108 0 0 1-1.646.005 1.108 1.108 0 0 1 .005-1.646l3.975-4.008-3.974-3.974a1.108 1.108 0 0 1-.006-1.646 1.108 1.108 0 0 1 1.646.005l4.008 3.975 3.974-3.974a1.109 1.109 0 0 1 1.646-.006c.486.45.484 1.198-.006 1.646l-3.974 3.974 3.975 4.008c.489.448.49 1.196.005 1.646Z"
        clipRule="evenodd"
      />
    </Svg>
  );
};
const Button = memo(SvgComponent);
export default Button;
