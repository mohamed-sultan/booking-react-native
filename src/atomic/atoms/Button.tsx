import React from 'react';
import {TouchableOpacity, Text, StyleSheet} from 'react-native';
import  { useThemeColors } from '@store/useTheme';

const Button = ({title, onPress, style, textStyle, disabled}: any) => {
  const colors = useThemeColors();
  const styles = getStyles(colors);
  return (
    <TouchableOpacity
      disabled={disabled}
      style={[styles.button, style, disabled && styles.disabledButton]}
      onPress={onPress}>
      <Text style={[styles.buttonText, textStyle]}>{title}</Text>
    </TouchableOpacity>
  );
};

export default React.memo(Button);

const getStyles = (colors: any) => StyleSheet.create({
  button: {
    backgroundColor: colors.PrimaryBackground,
    paddingVertical: 14,
    borderRadius: 8,
    alignItems: 'center',
  },
  buttonText: {
    color: 'white',
    fontSize: 16,
    fontWeight: '600',
  },
  disabledButton: {
    backgroundColor: colors.PrimaryBackground,
    opacity: 0.5,
  },
});
