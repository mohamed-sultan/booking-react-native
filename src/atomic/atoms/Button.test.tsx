import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import Button from './Button';

// Mock the useThemeColors hook
jest.mock('@store/useTheme', () => ({
  useThemeColors: () => ({
    PrimaryBackground: '#6200ee',
  }),
}));

describe('Button', () => {
  it('renders correctly with given title', () => {
    const { getByText, toJSON } = render(
      <Button title="Press me" onPress={jest.fn()} />
    );
    expect(getByText('Press me')).toBeTruthy();
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const { toJSON } = render(
      <Button title="Disabled" onPress={jest.fn()} disabled />
    );
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when pressed', () => {
    const onPressMock = jest.fn();
    const { getByText } = render(<Button title="Click me" onPress={onPressMock} />);
    fireEvent.press(getByText('Click me'));
    expect(onPressMock).toHaveBeenCalledTimes(1);
  });
});
