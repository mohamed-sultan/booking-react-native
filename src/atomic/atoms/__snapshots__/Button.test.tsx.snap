// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly when disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6200ee",
      "borderRadius": 8,
      "marginTop": 32,
      "opacity": 0.5,
      "paddingVertical": 14,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        },
        undefined,
      ]
    }
  >
    Disabled
  </Text>
</View>
`;

exports[`Button renders correctly with given title 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#6200ee",
      "borderRadius": 8,
      "marginTop": 32,
      "opacity": 1,
      "paddingVertical": 14,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "white",
          "fontSize": 16,
          "fontWeight": "600",
        },
        undefined,
      ]
    }
  >
    Press me
  </Text>
</View>
`;
